dialer.go

```diff
@@ -60,6 +61,9 @@ type Dialer struct {
 	// TLS enables Dialer to open secure connections.  If nil, standard net.Conn
 	// will be used.
 	TLS *tls.Config
+
+	// SASLClient enables SASL authentication
+	SASLClient SASLClient
```

Comment:
Might need to be a "factory" to prevent issues with reuse and concurrency:
```suggestion
	SASLClient func() SASLClient
```